core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -104,6 +104,15 @@ private[deploy] class Worker(
   private val CLEANUP_NON_SHUFFLE_FILES_ENABLED =
     conf.get(config.STORAGE_CLEANUP_FILES_AFTER_EXECUTOR_EXIT)
 
+  val EXTERNAL_SHUFFLE_SERVICE_ENABLED = conf.get(config.SHUFFLE_SERVICE_ENABLED)
+
+  if (CLEANUP_NON_SHUFFLE_FILES_ENABLED && EXTERNAL_SHUFFLE_SERVICE_ENABLED) {
```

Comment:
Are you sure that there isn't other stuff you can delete that are neither shuffle nor cached files?

If really there isn't anything else to clean, it seems that now both of these config options are basically the same thing.

If there is something, then that should still be done, otherwise you'll be using disk space unnecessarily.

(There's a whole other discussion about how to even delete orphaned shuffle and cached files after the executor that created them goes away, but that's for a separate discussion.)

Reply:
Good point. I am sure there are other files based on the different BlockIds we have:
https://github.com/apache/spark/blob/5933ef0723539807356153c767696a1d3cd2b144/core/src/main/scala/org/apache/spark/storage/BlockId.scala#L105-L114

Like temporary blocks. And I am uncertain whether all of these files have a proper cleanup (like `RemoteBlockDownloadFileManager`). So to be on the safe side I will create a new cleanup function which deletes non-shuffle and non-cached files only and in this case I will use this new cleanup method.